### PR TITLE
Feature/auto sort

### LINKED ISSUE #161: Auto-sort task lists
Event > Deadline > Float
Compare duedate, enddate, and startdate for same types.
If all tie, sort in alphabetical order.

### PR DESCRIPTION
**Pull request by bsmmoon**
It closes #161

### CODE REVIEW

File: DoLah/Models/Calendar.h [hunk 1]
```diff
@@ -20,7 +22,10 @@ namespace DoLah {
         void setDoneTask(int, bool=true);
 		void updateTask(int, AbstractTask*);
 		void clearTasks();
-		void search(std::string);
+        void search(std::string);
+        void sortTasks(std::vector<AbstractTask*>&);
+        static bool taskCompare(AbstractTask*, AbstractTask*);
+        static std::vector<std::tm> getDates(AbstractTask*);
```
Comment (9gix) [inline DoLah/Models/Calendar.h]: This function kills the whole idea of polymorphism.

Comment (bsmmoon) [inline DoLah/Models/Calendar.h]: There are many functions here, so please elaborate.

Comment (9gix) [inline DoLah/Models/Calendar.h]: I'm commenting at line 28

Comment (bsmmoon) [inline DoLah/Models/Calendar.h]: ok then please elaborate further. What you are disagree and a potential alternative and etc etc.

Comment (9gix) [inline DoLah/Models/Calendar.h]: Actually, Dynamic Casting also an anti-pattern of polymorphism.

It seems that there is a need to rethink about the design of this `FloatingTask`, `EventTask` and `DeadlineTask`

Comment (bsmmoon) [inline DoLah/Models/Calendar.h]: IFF you have the time to make sure everything is settled by Tuesday. 0.4 is the final version as you know very well.

**Comment by 9gix** [comment]
From Messenger
> [10/31/2015, 9:50 PM] @eatedcookie : @sm can you change sorting to mix events and deadlines?
> [10/31/2015, 9:50 PM] @bsmmoon : can be done. just need a clear ordering rule.
> [10/31/2015, 9:50 PM] @eatedcookie : events dont necessarily have priority over deadlines so just sort the both together by time

**Comment by eatedcookie** [comment]
got Expression: Invalid Comparator error and cannot run program. fix postpone to v0.5?

**Pull request merged**